Sources/NIO/SelectableEventLoop.swift

```diff
@@ -0,0 +1,455 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Dispatch
+import NIOConcurrencyHelpers
+
+/// `EventLoop` implementation that uses a `Selector` to get notified once there is more I/O or tasks to process.
+/// The whole processing of I/O and tasks is done by a `NIOThread` that is tied to the `SelectableEventLoop`. This `NIOThread`
+/// is guaranteed to never change!
+@usableFromInline
+internal final class SelectableEventLoop: EventLoop {
```

Comment:
this stuff is just moved apart from `s/public/internal` and fallout like `@usableFromInline` etc.